ParquetClassLibrary/Crafting/CraftingRecipe.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Collections.Generic;
+using ParquetClassLibrary.Sandbox.ID;
+
+namespace ParquetClassLibrary.Crafting
+{
+    /// <summary>
+    /// Models the ingredients and process needed to produce a new item.
+    /// </summary>
+    public struct CraftingRecipe
+    {
+        #region Defaults
+        /// <summary>Maximum width of the <see cref="StrikePanel"/> patten.</summary>
+        public const int PanelPatternMaxWidth = 2;
+
+        /// <summary>Maximum height of the <see cref="StrikePanel"/> patten.</summary>
+        public const int PanelPatternMaxHeight = 4;
+        #endregion
+
+        /// <summary>The type of item created by following this recipe.</summary>
+        public EntityID ItemProduced { get; }
+
+        /// <summary>The number of items created by following this recipe.</summary>
+        public int QuantityProduced { get; }
+
+        /// <summary>All materials needed to follow this recipe once.</summary>
+        public readonly List<EntityID> Ingredients;
+
+        /// <summary>The arrangment of panels encompassed by this recipe.</summary>
+        public readonly StrikePanel[,] PanelPattern;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CraftingRecipe"/> struct.
+        /// </summary>
+        /// <param name="in_itemProduced">The type of item created by following this recipe.</param>
+        /// <param name="in_quantityProduced">The number of items created by following this recipe.</param>
+        /// <param name="in_ingredients">All items needed to follow this recipe once.</param>
+        /// <param name="in_panelPattern">The arrangment of panels encompassed by this recipe.</param>
+        /// <exception cref="ArgumentOutOfRangeException">
```

Comment:
I'm normally not a fan of listing exceptions thrown in the constructor (feels java-y), but perhaps the IndexOutOfRangeException could be described here too?

Reply:
Yes, it should have been.  Thanks for catching this omission.